fabric-network/src/impl/query/queryagent.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2020 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict';
+
+const logger = require('fabric-network/lib/logger').getLogger('QueryAgent');
+
+/**
+ * @typedef {Object} QueryAgent~QueryResponse
+ * @memberof module:fabric-network
+ * @property {number} status - The status value from the endorsement. This attriibute
+ * will be set by the chaincode.
+ * @property {Buffer} payload - The payload value from the endorsement. This attribute
+ * may be considered the query value if the status code is exceptable.
+ * @property {Buffer} payload - The message value from the endorsement. This attribute
+ * will have a value when there is not a payload and status value indicates an issue
+ * with determining the payload (the query value).
+ */
+/**
+ * Used by query handler implementations to evaluate transactions on peers of their choosing.
+ * @memberof module:fabric-network
+ */
+class QueryAgent {
```

Comment:
Can we keep it as `Query` (at least in the TypeScript and JSDoc) to be consistent with v1.4 of the Node SDK and all versions of the Java SDK? If you want to keep this implementation called `QueryAgent` then we could define and document `Query` as a TypeScript interface.

Reply:
fix